tdrs-backend/tdpservice/parsers/fields.py

```diff
@@ -0,0 +1,86 @@
+"""Datafile field representations."""
+from .util import value_is_empty
+
+
+class Field:
+    """Provides a mapping between a field name and its position."""
+
+    def __init__(self, item, name, type, startIndex, endIndex, required=True, validators=[]):
+        self.item = item
+        self.name = name
+        self.type = type
+        self.startIndex = startIndex
+        self.endIndex = endIndex
+        self.required = required
+        self.validators = validators
+
+    def create(self, item, name, length, start, end, type):
+        """Create a new field."""
+        return Field(item, name, type, length, start, end)
+
+    def __repr__(self):
+        """Return a string representation of the field."""
+        return f"{self.name}({self.startIndex}-{self.endIndex})"
+
+    def parse_value(self, line):
+        """Parse the value for a field given a line, startIndex, endIndex, and field type."""
+        value = line[self.startIndex:self.endIndex]
+
+        if value_is_empty(value, self.endIndex-self.startIndex):
+            return None
+
+        match self.type:
+            case 'number':
+                try:
+                    value = int(value)
+                    return value
+                except ValueError:
+                    return None
+            case 'string':
+                return value
+
+
+def is_encrypted(value, decryption_dict):
+    """Determine if value is encrypted."""
+    return len([x for x in value if x in decryption_dict]) == 9
```

Comment:
minor note that this doesn't try to use the header `E` value; should we have some validation along those lines somewhere? cc @ADPennington

Reply:
Ya that is a nice catch! I didnt think of that case. We do need to check if the header has the encryption indicator then. I will make those changes now and ping you when they are ready for your eyes again. Also, your conclusions on the checks are all correct.